src/rpc/server.cpp

```diff
@@ -425,6 +426,21 @@ static inline JSONRPCRequest transformNamedArguments(const JSONRPCRequest& in, c
         } else {
             hole += 1;
         }
+        if (out.params.size() == 0) skipped += 1;
+    }
+    // If leftover "args" param was found, use it as a source of positional
```

Comment:
It seems to me that it would be clearer to check for `args` first and then fill in the named parameters afterwards. Overall, I think the logic would be easier to follow. Additionally, I think it would be clearer as to how we detect that an argument was both positional and named. The current method of detecting that is confusing to me.